contracts/base/ExecutionHelper.sol

```diff
@@ -122,6 +147,21 @@ contract ExecutionHelper is IExecutionHelperEventsAndErrors {
         }
     }
 
+    /// @dev Execute a delegatecall with `delegate` on this account.
+    /// Same as _executeDelegatecall but without return data for gas optimization.
+    function _executeDelegatecallNoReturndata(address delegate, bytes calldata callData) internal {
+        assembly {
+            let result := mload(0x40)
+            calldatacopy(result, callData.offset, callData.length)
+            if iszero(delegatecall(gas(), delegate, result, callData.length, codesize(), 0x00)) {
+                // Bubble up the revert if the call reverts.
+                returndatacopy(result, 0x00, returndatasize())
+                revert(result, returndatasize())
+            }
+            mstore(0x40, add(result, callData.length)) //allocate memory
```

Comment:
verify logic

Reply:
can you elaborate please?

Reply:
couldn't digest size of the allocated memory will be callData.length